common/backoff/cron.go

```diff
@@ -58,8 +58,14 @@ func GetBackoffForNextSchedule(cronSchedule string, startTime time.Time, closeTi
 	if err != nil {
 		return NoBackoff
 	}
+
+	if closeTime.Before(startTime) {
```

Comment:
I don't believe this is the right fix.  If startTime comes after closeTime, then it means some other event triggered invocation of this code path like workflow timeout.  In this case we should still try to fire the cron on previous value so we should just return start the delta between startTime and closeTime immediately.